### PR TITLE
TM-4137: export agendas by panel meeting

### PR DESCRIPTION
**Pull request by pveland621**
ACs:
1. Verify new export agendas by panel id EP is functional and export file is accurate

Dual Merge: 
- [FE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2481)

### CODE REVIEW

File: talentmap_api/fsbid/urls/panel.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
     url(r'^meetings/export/$', views.PanelMeetingsCSVView.as_view(), name="panel-meetings-export"),
     url(r'^meetings/', views.PanelMeetingsView.as_view(), name="panel-meetings-list"),
     url(r'^(?P<pk>[0-9]+)/agendas/', agenda_views.PanelAgendasListView.as_view(), name="panel-agendas-list"),
+    url(r'^(?P<pk>[0-9]+)/export/', agenda_views.PanelAgendasCSVView.as_view(), name="panel-agendas-export"),
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/urls/panel.py:17]: let's match this to  the `agendas` one, so `r'^(?P<pk>[0-9]+)/agendas/export/'`

File: talentmap_api/fsbid/services/agenda.py [hunk 2]
```diff
@@ -663,3 +668,71 @@ def get_agendas_by_panel(pk, jwt_token):
     )
 
     return agendas_by_panel
+
+def get_agendas_by_panel_export(pk, jwt_token, host=None):
+    print('agenda services pk: ', pk)
```
[the remark this answers is not in the document]
Comment (pveland621) [reply talentmap_api/fsbid/services/agenda.py:673]: sure didn't!

File: talentmap_api/fsbid/services/common.py [hunk 3]
```diff
@@ -973,6 +973,9 @@ def csv_fsbid_template_to_tm(data, mapping):
 
     return row
 
+def process_dates_csv(date):
+    return maya.parse(date).datetime().strftime('%m/%d/%Y')
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/common.py:977]: 🙌

File: talentmap_api/fsbid/views/agenda.py [hunk 4]
```diff
@@ -149,3 +149,12 @@ def get(self, request, pk):
         Get agendas for a panel meeting
         '''
         return Response(services.get_agendas_by_panel(pk, request.META['HTTP_JWT']))
+
+class PanelAgendasCSVView(BaseView):
+    permission_classes = [Or(isDjangoGroupMember('cdo'), isDjangoGroupMember('ao_user'),)]
+
+    def get(self, request, pk):
+        '''
+        Get agendas for a panel meeting
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/views/agenda.py:158]: for export

File: talentmap_api/fsbid/services/agenda.py [hunk 5]
```diff
@@ -663,3 +668,71 @@ def get_agendas_by_panel(pk, jwt_token):
...
+    '''
+    Get agendas for panel meeting
+    '''
+    mapping_subset = {
+        'default': 'None Listed',
+        'wskeys': {
+            'agendaAssignment[0].position[0].postitledesc': {},
+            'agendaAssignment[0].position[0].posnumtext': {},
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/agenda.py:681]: for position number and grade, you'll want something like `smart_str("=\"%s\"" % value)`

File: talentmap_api/fsbid/urls/panel.py (hunk 1 above)
Comment (pveland621) [reply talentmap_api/fsbid/urls/panel.py]: implemented this, but again seeing the weird issue where if i put the new export url below `r'^(?P<pk>[0-9]+)/agendas'` the export url is never hit. i put it above the other url just like before and that fixes it but still just strange

Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/urls/panel.py]: 🤷‍♀️ good to know!

File: talentmap_api/fsbid/views/agenda.py (hunk 4 above)
Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/views/agenda.py:158]: can we change this to a description that says export?

File: talentmap_api/fsbid/services/agenda.py [hunk 6]
```diff
@@ -663,3 +668,74 @@ def get_agendas_by_panel(pk, jwt_token):
...
+            'agendaAssignment[0].position[0].posorgshortdesc': {},
+            'agendaAssignment[0].asgdetadate': {
+                'transformFn': services.process_dates_csv,
+            },
+            'agendaAssignment[0].asgdetdteddate': {
+                'transformFn': services.process_dates_csv,
+            },
+            'agendaAssignment[0].asgdtoddesctext': {},
+            'agendaAssignment[0].position[0].posgradecode': {
+                'transformFn': lambda x : smart_str("=\"%s\"" % x),
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/agenda.py:692]: beautiful

**Review by elizabeth-jimenez — APPROVED** [review]
🦑

**Comment by elizabeth-jimenez** [comment]
@pveland621 approved, but can we please update that description? https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/767#discussion_r1131057971

**Pull request merged**